lattices/src/ght_macros.rs

```diff
@@ -0,0 +1,107 @@
+#[macro_export]
+/// Internal macro for constructing a Ght struct with the given schema and storage type
+///
+/// Should not be used directly, use `GhtType!` instead
+macro_rules! GhtTypeWithSchema {
+    // Empty key & Val (Leaf)
+    (() => () => $( $schema:ty ),+ : $storage:ident) => (
+        $crate::ght::GhtLeaf::<$( $schema ),*,  ()  >
+    );
+
+    // Empty key (Leaf)
+    (() => $( $z:ty ),* => $schema:ty : $storage:ident) => (
+        $crate::ght::GhtLeaf::<$schema,  $crate::variadics::var_type!($( $z ),*), $crate::variadics::variadic_collections::$storage<$schema> >
+    );
+
+    // Singleton key & Empty val (Inner over Leaf)
+    ($a:ty => () => $schema:ty : $storage:ident) => (
+        $crate::ght::GhtInner::<$a, $crate::ght::GhtLeaf::<$schema, (), $crate::variadics::variadic_collections::$storage<$schema> >>
+    );
+
+    // Singleton key (Inner over Leaf)
+    ($a:ty => $( $z:ty ),* => $schema:ty : $storage:ident) => (
+        $crate::ght::GhtInner::<$a, $crate::ght::GhtLeaf::<$schema, $crate::variadics::var_type!($( $z ),*), $crate::variadics::variadic_collections::$storage<$schema> >>
+    );
+
+    // Recursive case with empty val
+    ($a:ty, $( $b:ty ),* => () => $schema:ty : $storage:ident) => (
+        $crate::ght::GhtInner::<$a, $crate::GhtTypeWithSchema!($( $b ),* => () => $schema : $storage)>
+    );
+
+    // Recursive case
+    ($a:ty, $( $b:ty ),* => $( $z:ty ),* => $schema:ty : $storage:ident) => (
+        $crate::ght::GhtInner::<$a, $crate::GhtTypeWithSchema!($( $b ),* => $( $z ),* => $schema : $storage)>
+    );
+}
+
+#[macro_export]
+/// Public macro for constructing a Ght struct with the given schema and storage type
+///
+/// # Example
+/// ```
+/// use lattices::GhtType;
+/// use variadics::variadic_collections::VariadicHashSet;
+///
+/// type MyGht = GhtType!(u16, u32 => u64: VariadicHashSet);
+/// ```
```

Comment:
Add a note about what happens without `=>` (is that equivalent to `=> ()`?)

Reply:
it's an error with the `=>`. I updated the docs/examples a bit,